manifests/centosimage.pp

```diff
@@ -29,18 +30,32 @@
     before  => Docker::Image['centosagent']
   }
 
+  # Mount the gem cache directory because docker doesn't follow symlinks
+  # mount --bind /var/cache/rubygems rubygems
+  file {'/var/docker/centosagent/rubygems':
+    ensure => directory,
+    before => Mount['/var/docker/centosagent/rubygems'],
+  }
+
+  mount {'/var/docker/centosagent/rubygems':
```

Comment:
This is what I'm talking about: https://github.com/puppetlabs/puppetfactory/blob/master/puppetfactory/lib/puppetfactory/plugins/docker.rb#L33-L37